File: python/day01.py
def solution():
    data = open("../data/01.txt").read()
    cals = [[int(y) for y in x.split("\n")]
            for x in data.split("\n\n")]

    res1 = max(sum(s) for s in cals)
    res2 = sum(sorted(sum(s) for s in cals)[-3:])

    print(f'Answer 1: {res1}\nAnswer 2: {res2}')

File: rust/src/day01.rs
extern crate kth;

use kth::SliceExtKth;

use crate::common;

pub(crate) fn solution() {
    let data = common::read_lines(&common::data_file(1)).unwrap();
    let cals: Vec<Vec<i64>> = data.into_iter().fold(Vec::new(), |mut res, x| {
        if x == "" || res.is_empty() {
            res.push(Vec::new());
        } else {
            res.last_mut().unwrap().push(x.parse().unwrap());
        }
        res
    });
    let mut sums: Vec<i64> = cals.iter().map(|x| x.iter().sum()).collect();
    let res1: i64 = *sums.iter().max().unwrap();
    let n = sums.len();
    sums.partition_by_kth(n - 3);
    let res2: i64 = sums.iter().rev().take(3).sum();
    println!("Answer 1: {}, Answer 2: {}", res1, res2);
}
